services/apps/src/apps/service/apps/job_listings.clj

```diff
@@ -85,16 +85,17 @@
         search-params    (util/default-search-params params :startdate default-sort-dir)
         types            (.getJobTypes apps-client)
         jobs             (list-jobs* user search-params types analysis-ids)
+        rep-steps        (group-by (some-fn :parent-id :job-id) (jp/list-representative-job-steps (mapv :id jobs)))
```

Comment:
When I saw `some-fn` I couldn't help thinking of this:
http://imgs.xkcd.com/comics/cirith_ungol.png